implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs

```diff
@@ -66,22 +68,10 @@ impl CreateCommand {
 }
 
 async fn rpc(ctx: Context, (opts, mut cmd): (CommandGlobalOpts, CreateCommand)) -> Result<()> {
-    let lookup = opts.config.lookup();
     cmd.to = {
         let mut to = MultiAddr::default();
         for proto in cmd.to.iter() {
-            match proto.code() {
-                Node::CODE => {
-                    let alias = proto
-                        .cast::<Node>()
-                        .ok_or_else(|| anyhow!("invalid node address protocol"))?;
-                    let addr = lookup
-                        .node_address(&alias)
-                        .ok_or_else(|| anyhow!("no address for node {}", &*alias))?;
-                    to.try_extend(&addr)?
-                }
-                _ => to.push_back_value(&proto)?,
-            }
+            process_multi_addr(&proto, &mut to, &opts)?;
         }
         to
```

Comment:
The function you want to write should cover all the logic between lines 71 and 86 (both included). If you do that, the function signature would look like this:

```rust
pub fn process_multi_addr(addr: &MultiAddr, lookup: &ConfigLookup) -> anyhow::Result<()>
```

Now, to create test cases, you can create MultiAddr instances from plain strings, as well as for the ConfigLookup. This is a rough idea of what you could do: 

```rust
let lookup = |node: &str, addr: &str| {
    let mut l = ConfigLookup::new();
    l.set_node("n1", SocketAddr::from_str(addr).unwrap().into());
    l
};
let test_cases = vec![
    ("/node/n1", lookup("n1", "127.0.0.0:4000"), Ok(("n1")))
];
for (ma, lookup, expected) in test_cases {
    ...
}
```

Reply:
Thanks for the details!